tensorqtl/genotypeio.py

```diff
@@ -47,14 +47,21 @@ def __init__(self, generator, max_prefetch=10):
         self.start()
 
     def run(self):
-        for item in self.generator:
-            self.queue.put(item)
+        try:
+            for item in self.generator:
+                self.queue.put(item)
+        except Exception as exception:
+            self.queue.put(exception)
         self.queue.put(None)
 
     def next(self):
         next_item = self.queue.get()
         if next_item is None:
+            self.join()
```

Comment:
These `join` calls may not be needed but didn't seem to hurt anything. While researching the main thread getting stuck, I came across [this suggestion](https://github.com/justheuristic/prefetch_generator/issues/8#issuecomment-1857484127) that mentioned there may be memory leaks if `join` wasn't called.